packages/flutter/lib/src/gestures/long_press.dart

```diff
@@ -11,28 +11,50 @@ import 'recognizer.dart';
 /// same location for a long period of time.
 typedef GestureLongPressCallback = void Function();
 
+/// Signature for when a pointer stops to contacting the screen after a long period of time
+typedef GestureLongPressUpCallback = void Function();
+
 /// Recognizes when the user has pressed down at the same location for a long
 /// period of time.
 class LongPressGestureRecognizer extends PrimaryPointerGestureRecognizer {
   /// Creates a long-press gesture recognizer.
   ///
   /// Consider assigning the [onLongPress] callback after creating this object.
-  LongPressGestureRecognizer({ Object debugOwner }) : super(deadline: kLongPressTimeout, debugOwner: debugOwner);
+  LongPressGestureRecognizer({ Object debugOwner })
+      : super(deadline: kLongPressTimeout, debugOwner: debugOwner);
+
+  bool _longPressAccepted = false;
 
-  /// Called when a long-press is recognized.
+  /// Called when the pointer stops contacting the screen after the long-press gesture has been recognized.
   GestureLongPressCallback onLongPress;
 
+  /// Called when the pointer stops contacting the screen after the long-press gesture has been recognized
+  GestureLongPressUpCallback onLongPressUp;
+
   @override
   void didExceedDeadline() {
     resolve(GestureDisposition.accepted);
-    if (onLongPress != null)
+    _longPressAccepted = true;
+    if (onLongPress != null) {
       invokeCallback<void>('onLongPress', onLongPress);
+    }
   }
 
   @override
   void handlePrimaryPointer(PointerEvent event) {
-    if (event is PointerUpEvent)
-      resolve(GestureDisposition.rejected);
+    if (event is PointerUpEvent) {
+      if (_longPressAccepted == true && onLongPressUp != null) {
+        _longPressAccepted = false;
+        invokeCallback<void>('onLongPressUp', onLongPressUp);
+      } else {
+        resolve(GestureDisposition.rejected);
+      }
+    }
+
+    else if (event is PointerDownEvent) {
```

Comment:
should we set _longPressAccepted to null if the event is PointerCancelEvent?

Reply:
I feel like that makes sense!

Reply:
oops I meant false not null